packages/compass-shell/src/stores/store.js

```diff
@@ -47,21 +54,17 @@ export default class CompassShellStore {
     this.reduxStore.dispatch(globalAppRegistryActivated(appRegistry));
   }
 
+  onEnableShellChanged = (value) => {
+    this.reduxStore.dispatch(changeEnableShell(value));
+  };
+
   onDataServiceConnected = (error, dataService) => {
     this.reduxStore.dispatch(
       setupRuntime(error, dataService, this.globalAppRegistry)
     );
   };
 
   onDataServiceDisconnected = () => {
-    const {
-      runtime: { runtime },
-    } = this.reduxStore.getState();
-
-    if (runtime) {
-      runtime.terminate();
-    }
-
     this.reduxStore.dispatch(setupRuntime(null, null, null));
```

Comment:
Do we ever unset the `dataService` on the store?
How do you feel about adding a new redux action for this, that way the reducer then responds to the disconnect action by terminating the runtime if it exists.

Here's how we do something similar in `import-export`:
Calling the action:
https://github.com/mongodb-js/compass/blob/8a7cf9d3bb953cc4b8bbbf8c417575fd4b33eb34/packages/compass-import-export/src/stores/export-store.ts#L37
Here's one place where it's handled:
https://github.com/mongodb-js/compass/blob/8a7cf9d3bb953cc4b8bbbf8c417575fd4b33eb34/packages/compass-import-export/src/modules/export.ts#L624
Then it's unset on the store with the same action here:
https://github.com/mongodb-js/compass/blob/8a7cf9d3bb953cc4b8bbbf8c417575fd4b33eb34/packages/compass-import-export/src/modules/compass/data-service.ts#L49

`dataService` is on this same reducer so we could also unset it when we take down the runtime.

Reply:
Maybe I’m missing something, but this here is the exact line that creates the redux action to unset `dataService` on the store and have that call `state.runtime.terminate()` … we could also send another action instead of `setupRuntime(null, null, null)`, I don’t know what the best practices for redux are, but it doesn’t seem to make a practical difference?

Reply:
Yup no difference, fine as is